libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -525,6 +525,25 @@ bool AP_GPS::vertical_accuracy(uint8_t instance, float &vacc) const
     return false;
 }
 
+AP_GPS::GPS_Covariance_Type AP_GPS::position_covariance(const uint8_t instance, Matrix3d& cov) const
```

Comment:
Does this really need to be double?

Reply:
Yes, that's what ROS exposes. I could template it if you want.

Reply:
I ended up converting to float in fdd56d5c69477e1c02886d96f36deb4282276c1b.